src/tools/singlejar/port.h

```diff
@@ -36,7 +49,7 @@
 #define fstat _fstat64
 #endif  // _WIN64
 
-typedef ptrdiff ssize_t;
+typedef ptrdiff_t ssize_t;
```

Comment:
WDYT about using [`std::ptrdiff_t`](https://en.cppreference.com/w/cpp/types/ptrdiff_t)? I'd ~hope~ expect the STL types are more likely to be the same on all platforms than the POSIX types.

Reply:
C++ `std::ptrdiff_t` is the same as C `ptrdiff_t`. It is implemented as `namespace std { using ::ptrdiff_t; }` in `<cstddef>`.